src/gluonts/transform/sampler.py

```diff
@@ -142,6 +162,37 @@ def __call__(self, ts: np.ndarray) -> np.ndarray:
         return indices + a
 
 
+class ExpectedNumInstanceWithMinSampler(ExpectedNumInstanceSampler):
```

Comment:
I see two ways to escape the naming dilemma here:
1. Make `min_instances: int = 0` (note the default `0`) an attribute of `ExpectedNumInstanceSampler` and add the logic directly there. Then there's no name to be invented. However, strictly speaking when one sets `min_instances > 0` then this is not really sampling `num_instances` on average.
2. Make this logic a wrapper around some other sampler: `sampler = AtLeast(1, ExpectedNumInstanceSampler)`. The logic is factored a bit better maybe that way, just not as straightforward to instantiate.

Again, no strong opinions, it's just that this names is now a mouthful. @jaheba to the rescue.

Reply:
@abdulfatir let's go for option 1